src/cortex_cli/cortex_cli.py

```diff
@@ -193,6 +201,73 @@ def _validate_tokens_file(tokens_file: str) -> TokensFile:
     return validated_tokens
 
 
+def _validate_auth_server_url(ctx: click.Context, param: click.Option, base_url: str) -> str:
+    """Checks if provided auth server URL is valid, i.e. it:
+       - is a valid HTTP/HTTPS URL
+       - is accessible
+       - points to an authentication server
+
+    Args:
+        ctx: click context
+        param: click prompt param object
+        base_url (str): auth server base URL to validate
+    Returns:
+        str: validated auth server base URL
+    """
+    if ctx.obj and param.name in ctx.obj:
+        return base_url
+
+    is_valid = False
+    while not is_valid:
+        try:
+            master = requests.get(f'{base_url}/realms/master', timeout=AUTH_REQUESTS_TIMEOUT)
+            assert master.status_code == 200
+            assert 'public_key' in master.json()
+            is_valid = True
+        except (ConnectionError, AssertionError, ValueError):
```

Comment:
I believe `master.json()` would fail with `JSONDecodeError` if `master` is not JSON.

Reply:
JSONDecodeError is a subclass of ValueError, so it is already covered.